src/fstext/lattice-weight.h

```diff
@@ -330,6 +330,34 @@ class NaturalLess<LatticeWeightTpl<FloatType> > {
     return (Compare(w1, w2) == 1);
   }
 };
+template<>
+class NaturalLess<LatticeWeightTpl<float> > {
+ public:
+  typedef LatticeWeightTpl<float> Weight;
+
+  NaturalLess() {}
```

Comment:
This ever-multiplying constructor is not needed.

Reply:
I'm adressing the changes from @kkm000 
Removing the constructor, howevcer, is not completely trivial -- after removal, I get the following errors:
```
g++ -std=c++11 -I.. -I/export/a09/jtrmal/kaldi-clean-git/tools/openfst/include  -Wall -Wno-sign-compare -Wno-unused-local-typedefs -Wno-deprecated-declarations -Winit-self -DKALDI_DOUBLEPRECISION=0 -DHAVE_EXECINFO_H=1 -DHAVE_CXXABI_H -DHAVE_ATLAS -I/export/a09/jtrmal/kaldi-clean-git/tools/ATLAS_headers/include -msse -msse2 -pthread -g  -fPIC -DHAVE_CUDA -I/usr/local/cuda/include   -c -o feature-mfcc.o feature-mfcc.cc
In file included from /export/a09/jtrmal/kaldi-clean-git/tools/openfst/include/fst/fst.h:27:0,
                 from /export/a09/jtrmal/kaldi-clean-git/tools/openfst/include/fst/expanded-fst.h:16,
                 from /export/a09/jtrmal/kaldi-clean-git/tools/openfst/include/fst/fstlib.h:28,
                 from ../fstext/fstext-lib.h:22,
                 from ../lat/kaldi-lattice.h:24,
                 from ../hmm/hmm-utils.h:25,
                 from hmm-utils.cc:22:
/export/a09/jtrmal/kaldi-clean-git/tools/openfst/include/fst/arc.h: In instantiation of ‘fst::ArcTpl<Weight>::ArcTpl() [with W = fst::LatticeWeightTpl<float>]’:
/export/a09/jtrmal/kaldi-clean-git/tools/openfst/include/fst/test-properties.h:113:11:   required from ‘uint64 fst::ComputeProperties(const fst::Fst<Arc>&, uint64, uint64*, bool) [with Arc = fst::ArcTpl<fst::LatticeWeightTpl<float> >; uint64 = long unsigned int]’
/export/a09/jtrmal/kaldi-clean-git/tools/openfst/include/fst/test-properties.h:216:50:   required from ‘uint64 fst::TestProperties(const fst::Fst<Arc>&, uint64, uint64*) [with Arc = fst::ArcTpl<fst::LatticeWeightTpl<float> >; uint64 = long unsigned int]’
/export/a09/jtrmal/kaldi-clean-git/tools/openfst/include/fst/fst.h:871:52:   required from ‘uint64 fst::ImplToFst<Impl, FST>::Properties(uint64, bool) const [with Impl = fst::internal::VectorFstImpl<fst::VectorState<fst::ArcTpl<fst::LatticeWeightTpl<float> >, std::allocator<fst::ArcTpl<fst::LatticeWeightTpl<float> > > > >; FST = fst::MutableFst<fst::ArcTpl<fst::LatticeWeightTpl<float> > >; uint64 = long unsigned int]’
/export/a09/jtrmal/kaldi-clean-git/tools/openfst/include/fst/vector-fst.h:430:7:   required from here
/export/a09/jtrmal/kaldi-clean-git/tools/openfst/include/fst/arc.h:38:12: error: no matching function for call to ‘fst::LatticeWeightTpl<float>::LatticeWeightTpl()’
   ArcTpl() {}
```

I'll look into it in the evening -- just letting you know I didn't abandon this.

Reply:
Wow! C++ loves us!